src/index.ts

```diff
@@ -27,7 +69,9 @@ class SocketIoConnection {
       | WebSocketServerConnectionProtocol,
   ) {}
 
-  public on(event: string, listener: BoundMessageListener): void {
+  public _onSocketIoPacket(
```

Comment:
If this is meant as an internal method, I recommend using `#` for the method name instead. That would truly make it private:

```suggestion
  #onSocketIoPacket(
```

The method could then be used as `this.#onSocketIoPacket()` within this class.

Reply:
I struggled with this because `SocketIoDuplexConnection.setAuthorizer()` has to use `SocketIoConnection._onSocketIoPacket()`.

We could move the connection acceptance logic to `SocketIoConnection` as in `io.client.setAuthorizer()` and then make `_onSocketIoPacket()` private. Then `sendMockEngineIoOpen` etc. could probably also move over to `SocketIoConnection` which seems conceptually better.